### ISSUE TITLE
Class Changing + Item Drop

### ISSUE THREAD

**Issue by netherfoam** [alkarinv/BattleArena#84]
I've mentioned it before somewhere, but I thought I'd make a ticket out of it:

When a player drops an item in a match, it would be nice to have a config option to:
1, Cancel the drop event (On second thought, isn't this itemDropOff?)
2, Allow class changing even if they've dropped an item

Cheers ^_^

**Comment by alkarinv** [comment]
Indeed itemDropOff, but its actually called..
dropItemOff, which I will have to switch around b/c itemDropOff makes a lot more sense.

I added in the arenas now no longer delete if they are bad, or have bad locations.

The changing class.. well I will have to change still

**Comment by alkarinv** [comment]
Well.. it will have to be an option for dropping and being allowed to change.. Problem is they can just drop all their items, then change.. and they basically have even more items.

**Comment by netherfoam** [comment]
EDIT: Derp, I didn't think that through very well.

Can you cancel the drop item event, and (since its cancelled) let them change class still?

**Comment by alkarinv** [comment]
Yes you can cancel the drop item event. "dropItemOff", but then they couldn't drop the items and could change class anyway?  I guess you just always want them to be able to change class?

**Comment by netherfoam** [comment]
Strange, I'm getting players who can't change classes with itemDropOff?

Also, where can I find a list of victoryConditions?
I'm using timeLimit, but I don't want the match to end unless all players are dead (=Loss) or my custom victory condition(=Win) - timeLimit causes the last man standing to win (?)

Because I'm trying to make a PvE arena using BattleArenas API

**Comment by alkarinv** [comment]
TimeLimit shouldn't cause the last man standing to win, it shouldn't cause anyone to win really.  All it should do is end the match after a certain amount of time.  Here is the current list of Victory Conditions.

https://github.com/alkarinv/BattleArena/tree/master/src/mc/alk/arena/objects/victoryconditions

I'll be trying to get all of those to work.  I will be changing nDeaths to NLives to make it easier to understand for people.

I'll work on the class stuff today.. I know there is some small issue there that prevents certain classes from switching to another class.

Oh, and it's dropItemOff, ItemDropOff will work only with the newest newest version not 3.7.8.4 not released yet.
